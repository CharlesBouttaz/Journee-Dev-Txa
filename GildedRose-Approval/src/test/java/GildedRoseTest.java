import java.util.ArrayList;
import java.util.List;

import org.approvaltests.Approvals;
import org.junit.Test;


public class GildedRoseTest {

	@Test
	public void goldenMasterShouldBeTheSame() throws Exception {
        final GildedRose gildedRose = new GildedRose();
        gildedRose.createItems();

        final List<String> res = new ArrayList<String>();

        for (int i = 0; i < 20; i++) {
            gildedRose.updateQuality();
            res.add(gildedRose.getItems().toString());
        }

        Approvals.verifyAll("Items", res.toArray());
    }

	@Test
	public void externalCallTest() throws Exception {
        final GildedRose gildedRose = new GildedRose();
        gildedRose.createItems();

        final List<String> res = new ArrayList<String>();

        for (int i = 0; i < 3; i++) {
            gildedRose.updateQuality();
            res.add(gildedRose.externalCallParams);
        }

        Approvals.verifyAll("Ordered Items", res.toArray());
    }

    public void parameterisedTests() {
        //LegacyApprovals.LockDown(object, "method", param1Range, param2range);
        // See https://www.youtube.com/watch?v=EJ1tRFEOkcw
    }

    public void binaryCompareTest() {
        // PDF
//        Approvals.verify(byte[])
//        Approvals.verify(inputStream)
    }
}
