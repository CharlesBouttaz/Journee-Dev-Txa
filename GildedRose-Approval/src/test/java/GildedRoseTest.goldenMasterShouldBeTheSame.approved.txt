Items[00] = [Item{name='+5 Dexterity Vest', sellIn=9, quality=19}, Item{name='Aged Brie', sellIn=1, quality=1}, Item{name='Elixir of the Mongoose', sellIn=4, quality=6}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=14, quality=21}, Item{name='Conjured Mana Cake', sellIn=2, quality=5}]
Items[01] = [Item{name='+5 Dexterity Vest', sellIn=8, quality=18}, Item{name='Aged Brie', sellIn=0, quality=2}, Item{name='Elixir of the Mongoose', sellIn=3, quality=5}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=13, quality=22}, Item{name='Conjured Mana Cake', sellIn=1, quality=4}]
Items[02] = [Item{name='+5 Dexterity Vest', sellIn=7, quality=17}, Item{name='Aged Brie', sellIn=-1, quality=4}, Item{name='Elixir of the Mongoose', sellIn=2, quality=4}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=12, quality=23}, Item{name='Conjured Mana Cake', sellIn=0, quality=3}]
Items[03] = [Item{name='+5 Dexterity Vest', sellIn=6, quality=16}, Item{name='Aged Brie', sellIn=-2, quality=6}, Item{name='Elixir of the Mongoose', sellIn=1, quality=3}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=24}, Item{name='Conjured Mana Cake', sellIn=-1, quality=1}]
Items[04] = [Item{name='+5 Dexterity Vest', sellIn=5, quality=15}, Item{name='Aged Brie', sellIn=-3, quality=8}, Item{name='Elixir of the Mongoose', sellIn=0, quality=2}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=25}, Item{name='Conjured Mana Cake', sellIn=-2, quality=0}]
Items[05] = [Item{name='+5 Dexterity Vest', sellIn=4, quality=14}, Item{name='Aged Brie', sellIn=-4, quality=10}, Item{name='Elixir of the Mongoose', sellIn=-1, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=9, quality=27}, Item{name='Conjured Mana Cake', sellIn=-3, quality=0}]
Items[06] = [Item{name='+5 Dexterity Vest', sellIn=3, quality=13}, Item{name='Aged Brie', sellIn=-5, quality=12}, Item{name='Elixir of the Mongoose', sellIn=-2, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=8, quality=29}, Item{name='Conjured Mana Cake', sellIn=-4, quality=0}]
Items[07] = [Item{name='+5 Dexterity Vest', sellIn=2, quality=12}, Item{name='Aged Brie', sellIn=-6, quality=14}, Item{name='Elixir of the Mongoose', sellIn=-3, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=7, quality=31}, Item{name='Conjured Mana Cake', sellIn=-5, quality=0}]
Items[08] = [Item{name='+5 Dexterity Vest', sellIn=1, quality=11}, Item{name='Aged Brie', sellIn=-7, quality=16}, Item{name='Elixir of the Mongoose', sellIn=-4, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=6, quality=33}, Item{name='Conjured Mana Cake', sellIn=-6, quality=0}]
Items[09] = [Item{name='+5 Dexterity Vest', sellIn=0, quality=10}, Item{name='Aged Brie', sellIn=-8, quality=18}, Item{name='Elixir of the Mongoose', sellIn=-5, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=35}, Item{name='Conjured Mana Cake', sellIn=-7, quality=0}]
Items[10] = [Item{name='+5 Dexterity Vest', sellIn=-1, quality=8}, Item{name='Aged Brie', sellIn=-9, quality=20}, Item{name='Elixir of the Mongoose', sellIn=-6, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=4, quality=38}, Item{name='Conjured Mana Cake', sellIn=-8, quality=0}]
Items[11] = [Item{name='+5 Dexterity Vest', sellIn=-2, quality=6}, Item{name='Aged Brie', sellIn=-10, quality=22}, Item{name='Elixir of the Mongoose', sellIn=-7, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=3, quality=41}, Item{name='Conjured Mana Cake', sellIn=-9, quality=0}]
Items[12] = [Item{name='+5 Dexterity Vest', sellIn=-3, quality=4}, Item{name='Aged Brie', sellIn=-11, quality=24}, Item{name='Elixir of the Mongoose', sellIn=-8, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=2, quality=44}, Item{name='Conjured Mana Cake', sellIn=-10, quality=0}]
Items[13] = [Item{name='+5 Dexterity Vest', sellIn=-4, quality=2}, Item{name='Aged Brie', sellIn=-12, quality=26}, Item{name='Elixir of the Mongoose', sellIn=-9, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=47}, Item{name='Conjured Mana Cake', sellIn=-11, quality=0}]
Items[14] = [Item{name='+5 Dexterity Vest', sellIn=-5, quality=0}, Item{name='Aged Brie', sellIn=-13, quality=28}, Item{name='Elixir of the Mongoose', sellIn=-10, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=0, quality=50}, Item{name='Conjured Mana Cake', sellIn=-12, quality=0}]
Items[15] = [Item{name='+5 Dexterity Vest', sellIn=-6, quality=0}, Item{name='Aged Brie', sellIn=-14, quality=30}, Item{name='Elixir of the Mongoose', sellIn=-11, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}, Item{name='Conjured Mana Cake', sellIn=-13, quality=0}]
Items[16] = [Item{name='+5 Dexterity Vest', sellIn=-7, quality=0}, Item{name='Aged Brie', sellIn=-15, quality=32}, Item{name='Elixir of the Mongoose', sellIn=-12, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}, Item{name='Conjured Mana Cake', sellIn=-14, quality=0}]
Items[17] = [Item{name='+5 Dexterity Vest', sellIn=-8, quality=0}, Item{name='Aged Brie', sellIn=-16, quality=34}, Item{name='Elixir of the Mongoose', sellIn=-13, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-3, quality=0}, Item{name='Conjured Mana Cake', sellIn=-15, quality=0}]
Items[18] = [Item{name='+5 Dexterity Vest', sellIn=-9, quality=0}, Item{name='Aged Brie', sellIn=-17, quality=36}, Item{name='Elixir of the Mongoose', sellIn=-14, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-4, quality=0}, Item{name='Conjured Mana Cake', sellIn=-16, quality=0}]
Items[19] = [Item{name='+5 Dexterity Vest', sellIn=-10, quality=0}, Item{name='Aged Brie', sellIn=-18, quality=38}, Item{name='Elixir of the Mongoose', sellIn=-15, quality=0}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-5, quality=0}, Item{name='Conjured Mana Cake', sellIn=-17, quality=0}]
